app/models/pages/frontmatter.rb

```diff
@@ -25,8 +31,15 @@ def perform_caching
       end
     end
 
-    def initialize(content_dir = nil)
-      @content_dir = content_dir ? Pathname.new(content_dir) : MARKDOWN_CONTENT_DIR
+    def initialize(content_dir_or_data = nil)
```

Comment:
This smells a bit to me - I may have misread but it looks like this has just been added for the test cases? I don't think we should be adding code to facilitate the test suite; could we create some mock markdown fixtures instead and use the existing initialiser?

Reply:
I was finding it difficult to test with and took the view if its hard to test than its probably assuming too much internally - hence allow for dependency injection

I can change this to a context instead though